code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,674 @@
+/obj/machinery/construction_nuke/proc/set_safety()
+	if(!anchored)
+		to_chat(usr, "<span class='danger'>Error: Safety cannot be altered on an unanchored device.</span>")
+		playsound(loc, 'sound/machines/defib_failed.ogg', 75, 1)
+		return
+	safety = !safety
+	if(safety)
+		if(timing)
+			priority_announce("Radioactive energy levels are normalizing, please submit an incident report as soon as possible.","Central Command Nuclear Safety Division", 'sound/AI/attention.ogg')
+		timing = FALSE
+		bomb_set = FALSE
+		detonation_timer = null
+		icon_state = "nuclearbomb2"
+		playsound(loc, 'sound/machines/terminal_prompt.ogg', 75, 1)
+	else
+		playsound(loc, 'sound/machines/engine_alert1.ogg', 50, 1)
+		icon_state = "nuclearbomb1"
+	update_icon()
+
+/obj/machinery/construction_nuke/proc/set_active()
+	var/area/A = get_area(src)
+	if(safety && !bomb_set)
+		to_chat(usr, "<span class='danger'>Error: The safety is still on.</span>")
+		playsound(loc, 'sound/machines/defib_failed.ogg', 75, 1)
+		return
+	if(!A.blob_allowed)
+		to_chat(usr, "<span class='danger'>Error: The device's safety countermeasures flash red: you cannot arm this device outside of the station.</span>")
+		playsound(loc, 'sound/machines/defib_failed.ogg', 75, 1)
+		return
+	timing = !timing
+	if(timing)
+		if(cooldown > world.time)
+			to_chat(usr, "<span class='danger'>Error: The device is still resetting from the last activation, it will be ready again in [round((cooldown-world.time)/10)] seconds.</span>")
+			playsound(loc, 'sound/machines/defib_failed.ogg', 75, 1)
+			timing = FALSE
+			return
+		bomb_set = TRUE
+		priority_announce("We are detecting a massive spike of radioactive energy from \a [payload] payload originating from [A.map_name]. If this is not a scheduled occurrence, please investigate immediately.","Nanotrasen Nuclear Safety Division", 'sound/misc/airraid.ogg')
+		cooldown = world.time + 1200
+		detonation_timer = world.time + (timer_set * 10)
+		icon_state = "nuclearbombc"
+	else
+		bomb_set = FALSE
+		priority_announce("Radioactive energy levels are normalizing, please submit an incident report as soon as possible.","Central Command Nuclear Safety Division", 'sound/AI/attention.ogg')
+		detonation_timer = null
+		icon_state = "nuclearbomb1"
+		playsound(loc, 'sound/machines/terminal_off.ogg', 75, 1)
+	update_icon()
+
+/obj/machinery/construction_nuke/proc/get_time_left()
+	if(timing)
+		. = round(max(0, detonation_timer - world.time) / 10, 1)
+	else
+		. = timer_set
+
+/obj/machinery/construction_nuke/proc/explode()
+	if(safety || !bomb_set)
+		timing = FALSE
+		return
+	exploding = TRUE
+	update_icon()
+	for(var/mob/M in player_list)
+		M << 'sound/machines/Alarm.ogg'
+	var/turf/startpoint = get_turf(src)
+	sleep(100)
+	for(var/mob/M in player_list)
+		M << 'sound/effects/explosionfar.ogg'
+	spawn_atom_to_turf(/obj/effect/overlay/temp/explosion, startpoint, 1, FALSE)
+	qdel(src)
+	for(var/I in spiral_range_turfs(range, startpoint))
```

Comment:
You need to cache payload_floor and payload_wall into proc vars before qdeling `src`